### PR TITLE
fix(security/helmet): fix import statement

### PR DESCRIPTION
**Pull request by stingalleman**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[X] Docs
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

`import * as helmet from 'helmet'` errors, it should be `import helmet from 'helmet'`. 

Error:
```
Type originates at this import. A namespace-style import cannot be called or constructed, and will cause a failure at runtime. Consider using a default import or import require here instead.
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by keesvv — APPROVED** [review]

File: content/security/helmet.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ $ npm i --save helmet
 Once the installation is complete, apply it as a global middleware.
 
 ```typescript
-import * as helmet from 'helmet';
+import helmet from 'helmet';
```
Comment (jmcdo29) [inline content/security/helmet.md:18]: Why the change? Using the newest version of `helmet` I'm able to use `import * as helmet from 'helmet'` without any warning or errors. Node version 14.5.0

Comment (stingalleman) [reply content/security/helmet.md:18]: Also on the latest version of `helmet`, node version 14.15.3 and the latest TypeScript SDK. I'm getting a compiler error: 
image

Comment (jmcdo29) [reply content/security/helmet.md:18]: Still not getting the same message to show up. 

Updated node to 14.15.3
Updated Typescript to 4.1.3
Updated helmet at 4.3.1
Using `import * as helmet from 'helmet'` without an issue. 

Same functionality with both `yarn` and `npm`

Comment (stingalleman) [reply content/security/helmet.md:18]: I don't know why it's working on your side but errors on mine, but I'm sure this is a good change. `Helmet` exports a default in `index.ts` (see [Helmet index.ts](https://github.com/helmetjs/helmet/blob/main/index.ts) and this 
image, it clearly shows that it's a default type)

Comment (jmcdo29) [reply content/security/helmet.md:18]: [Line 201](https://github.com/helmetjs/helmet/blob/main/index.ts#L201) `export = helmet`. There's no `default` there, and there's no default in the `d.ts`

Comment (jmcdo29) [reply content/security/helmet.md:18]: You _can_ do a `default` import (`import helmet from 'helmet'`) but it isn't neccessary

Comment (stingalleman) [reply content/security/helmet.md:18]: Yes, that's correct. The [TypeScript docs](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require) indicate that `export =` (in `helmet` index.ts) requires you to do `import helmet = require('helmet')`. That's the same as `import helmet from 'helmet'. Also see this [SO answer](https://stackoverflow.com/a/35706271), that confirms my claims.

Comment (stingalleman) [reply content/security/helmet.md:18]: > You _can_ do a `default` import (`import helmet from 'helmet'`) but it isn't neccessary

I missed this one. Yes, you can, but the SO answer linked above says this is not recommended.

> Attempting to use `import * as express` and then invoking `express()` is always illegal according to the ES6 spec. In some runtime+transpilation environments this might happen to work anyway, but it might break at any point in the future without warning, which will make you sad.

Comment (jmcdo29) [reply content/security/helmet.md:18]: Interesting. I wonder if some of this comes down to what `target` you're configured for

**Review by jmcdo29 — APPROVED** [review]
After the great discussion, I think should be a fine change to make. It's not that it's a default export, but an ES6 quirk/oddity. Maybe `import helmet = require('helmet')` would be better, but overall, I think this should be fine.

**Comment by stingalleman** [comment]
Just out of curiosity @jmcdo29, can you share the `tsconfig` you were using?

Mine: https://github.com/stingalleman/boilerplate.ts/blob/main/tsconfig.json

**Comment by jmcdo29** [comment]
@stingalleman just Nest's default tsconfig

```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "allowJs": true,
    "target": "es2017",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true
  }
}
```

**Comment by stingalleman** [comment]
@jmcdo29 I guess it has something to do with this, maybe the target. I was just curious why this was happening.

**Comment by kamilmysliwiec** [comment]
With `import helmet from 'helmet'`, the default Nest boilerplate (`typescript-starter` or this one generated by `nest new`) will crash with the following error:

```
(node:9173) UnhandledPromiseRejectionWarning: TypeError: helmet_1.default is not a function
```
To make it work properly, you must set the `esModuleInterop` to `true` in the `tsconfig.json` (it's disabled by default). 

On the other hand, when you have the `esModuleInterop` enabled, `import * as helmet` syntax won't work ("This expression is not callable" error). 

Would you like to update this PR with a hint (blockquote) that a different import statement must be used depending on the `esModuleInterop` configuration? @stingalleman

**Comment by stingalleman** [comment]
@kamilmysliwiec Pushed!

**Review by jmcdo29 — COMMENTED** [review]
I feel this should be the other way around. In the docs we keep the `import * as helmet from 'helmet'` and then mention the error you _might_ get with `esModuleInterop`, as a `nest new` project works with `import * as helmet from 'helmet'` and not `import helmet from 'helmet'`.

**Comment by stingalleman** [comment]
I agree, that's better.

**Comment by stingalleman** [comment]
Again, @jmcdo29, @kamilmysliwiec , I pushed

File: content/security/helmet.md [hunk 2]
```diff
@@ -20,6 +20,8 @@ import * as helmet from 'helmet';
 app.use(helmet());
 ```
 
+> info **Hint** If you are getting a error (`This expression is not callable`) while trying to import Helmet, you probably have `allowSyntheticDefaultImports` and `esModuleInterop` set to `true` in your project's `tsconfig.json`. If this is the case, you will need to use `import helmet from 'helmet'`, instead of the example.
```
Comment (kamilmysliwiec) [inline content/security/helmet.md:23]:
```suggestion
> info **Hint** If you are getting a `This expression is not callable` error while trying to import `Helmet`, you very likely have the `allowSyntheticDefaultImports` and `esModuleInterop` options set to `true` in your project's `tsconfig.json` file. If that's the case, change the import statement to: `import helmet from 'helmet'` instead.

File: content/security/helmet.md [hunk 3]
```diff
@@ -20,6 +20,8 @@ import * as helmet from 'helmet';
 app.use(helmet());
 ```
 
+> info **Hint** If you are getting a `This expression is not callable` error while trying to import `Helmet`, you very likely have the `allowSyntheticDefaultImports` and `esModuleInterop` options set to `true` in your project's `tsconfig.json` file. If that's the case, change the import statement to: `import helmet from 'helmet'` instead.
```
Comment (kamilmysliwiec) [inline content/security/helmet.md:23]:
```suggestion
> info **Hint** If you are getting the `This expression is not callable` error while trying to import `Helmet`, you very likely have the `allowSyntheticDefaultImports` and `esModuleInterop` options set to `true` in your project's `tsconfig.json` file. If that's the case, change the import statement to: `import helmet from 'helmet'` instead.
```

**Comment by kamilmysliwiec** [comment]
Thanks!
